src/components/manager/projects/RoutedProject.js

```diff
@@ -55,9 +64,8 @@ const RoutedProject = () => {
   }, []);
 
   const handleProjectSave = useCallback(
-    (newProject) => {
-      // TODO: Form validation for project
-      dispatch(saveProjectIfPossible(newProject));
+    async (newProject) => {
+      await dispatch(saveProjectIfPossible(newProject));
```

Comment:
this could just be `(newProject) => dispatch(saveProjectIfPossible(newProject))` to directly return the promise